Deprecate constructor and replace it by another one.

bangerth opened the pull request:
Also update the documentation in a number of places.

bangerth commented:
As found while looking over code with @ian-r-rose .

drwells commented on doc/news/changes.h:
```diff
@@ -139,6 +139,13 @@ inconvenience this causes.
 
 <ol>
+ <li> Changed: As MappingQEulerian before, MappingQ1Eulerian has gained
+ a second constructor that reverts the order of the arguments to indicate
```
Two not very important style comments:

1. the first clause is is a fragment (should be 'like MappingQEulerian', or 'Similiar to MappingQEulerian', or similar).
2. 'reverses' or 'switches', not 'reverts'

drwells commented:
I can't seem to click through to it, but there is a code example in the class documentation (lines 67 through 73) that uses the old constructor. Could you also update that?

Line 82 references `EulerVectorType` as the name of a template parameter: this should be `VectorType`. It is not related to the current PR, but you did fix some other small things in the  documentation.

drwells commented:
Clarification: the header file consistently uses `VectorType`, not `EulerVectorType`, as the template parameter, so the documentation is a bit misleading.

It looks like the source file uses `EulerVectorType`. I'll do a search and replace after this is merged.

bangerth commented:
All good comments. I've updated the pull request and also removed a comment in both classes that pertained to the copy constructor of `Vector` not being explicit, and that is no longer correct. It was also really difficult to understand if you didn't know the issue...

drwells commented:
Nice catch with the explicit thing.
